### PR TITLE
Create test app for staging website

### PR DESCRIPTION
**Pull request by scottnuma**
- Create a test application for the staging website that says `hello world!`
- We'd like to keep the staging website accessible but away from Google and the public, so I added http basic auth and a robots.txt
- No impact on the public website
- Contain all nginx configuration in /nginx/staging-nginx.conf

---------

- I configured the http basic auth with env vars at the start of runtime
- Alternatives considered
  - (1) could add the .htpasswd into the repo
    - This repo is public; the hashed password is public and crackable
  - (2) generate the .htpasswd during image build
    - Messy because then our image contains secrets
    - Environment variables and dockerfile ARGS are recoverable from the image

### CODE REVIEW

File: nginx/staging-nginx.conf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                      '"$http_user_agent" "$http_x_forwarded_for"';
+
+    access_log  /var/log/nginx/access.log  main;
+
+    sendfile        on;
+    #tcp_nopush     on;
+
+    keepalive_timeout  65;
+
+    #gzip  on;
```
Comment (quantumfusion) [inline nginx/staging-nginx.conf:29]: Probably fine to keep this on?

File: nginx/staging-nginx.conf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    auth_basic  "PiE Website Staging";
+    auth_basic_user_file "/usr/share/nginx/auth/.htpasswd";
+
+    include /etc/nginx/conf.d/*.conf;
```
Comment (quantumfusion) [inline nginx/staging-nginx.conf:34]: Do we still need this, or are we relying on a pre-baked config already in the docker image to host from that html dir?

File: Dockerfile.staging [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+COPY nginx/staging-nginx.conf /etc/nginx/nginx.conf
+COPY nginx/http-basic-auth-start.sh /etc/nginx/http-basic-auth-start.sh
+
+# Add website contents
+RUN echo "hello world!" > /usr/share/nginx/html/index.html
+
+# Prevent web scrapers
+RUN echo "User-agent: * Disallow: /" > /usr/share/nginx/html/robots.txt
+
+CMD /etc/nginx/http-basic-auth-start.sh
```
Comment (quantumfusion) [inline Dockerfile.staging:14]: Curious, is this being run as root, or does the nginx docker image provide an nginx user?

Comment (scottnuma) [reply Dockerfile.staging:14]: I think there's an nginx user
https://github.com/nginxinc/docker-nginx/blob/master/Dockerfile-debian.template#L10
I suspect this is the Dockerfile in use.

File: nginx/staging-nginx.conf (hunk 2 above)
Comment (scottnuma) [reply nginx/staging-nginx.conf:34]: This config is what the default config came with - I'm not too sure what the include actually picks up.

File: nginx/staging-nginx.conf (hunk 1 above)
Comment (scottnuma) [reply nginx/staging-nginx.conf]: Flipped on

File: nginx/staging-nginx.conf [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        location / {
+                root /usr/share/nginx/html;
+                index index.html; 
+        }
+
+        error_page   500 502 503 504  /50x.html;
+        location = /50x.html {
+            root   /usr/share/nginx/html;
+        }
+    }
```
Comment (scottnuma) [inline nginx/staging-nginx.conf:43]: This is a newly added section, that was previously in a linked file embedded within the nginx image.

File: nginx/staging-nginx.conf (hunk 2 above)
Comment (scottnuma) [reply nginx/staging-nginx.conf]: Removed

**Review by quantumfusion — APPROVED** [review]

**Pull request merged**